psydac/linalg/stencil.py

```diff
@@ -856,6 +894,28 @@ def dot( self, v, out=None):
         out.ghost_regions_in_sync = False
         return out
 
+    def vdot( self, v, out=None):
+
+        assert isinstance( v, StencilVector )
+        assert v.space is self.domain
+
+        if out is not None:
+            assert isinstance( out, StencilVector )
+            assert out.space is self.codomain
+        else:
+            out = StencilVector( self.codomain )
+
+        # Necessary if vector space is distributed across processes
+        if not v.ghost_regions_in_sync:
+            v.update_ghost_regions()
+
+        self._func(self._data, v._data.conjugate(), out._data, **self._args)
+        out._data = out._data.conjugate()
```

Comment:
Can you please explain what is happening here exactly? `v._data.conjugate()` creates a temporary Numpy array, and `out._data.conjugate()` creates another one. It should be possible to avoid both.

CC: @jowezarek

Reply:
[b643afc](https://github.com/pyccel/psydac/pull/273/commits/b643afc61c4cb72910d3f7e96bd91d81910fb207)